libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushMessaging.kt

```diff
@@ -109,41 +120,19 @@ object PushMessaging {
 
             // Deletes InstanceID only if there are no other logged in accounts.
             if (isLastAccount) {
-                initializeFirebaseIfNeeded(context)
-                val appName = getAppNameForFirebase(context)
-                val instanceID = FirebaseInstanceId.getInstance(FirebaseApp.getInstance(appName))
-                kotlin.runCatching { instanceID.deleteInstanceId() }
+                val firebaseApp = getFirebaseApp(context)
+                val firebaseMessaging = SalesforceSDKManager.getInstance()
+                    .pushNotificationReceiver?.supplyFirebaseMessaging() ?: FirebaseMessaging.getInstance()
+                firebaseMessaging.isAutoInitEnabled = false
+                FirebaseInstallations.getInstance(firebaseApp).delete()
+                firebaseApp.delete()
             }
+            unregisterSFDCPush(context, account)
         }
 
         unregisterSFDCPush(context, account)
     }
 
-    /**
-     * Will make call to Firebase.initializeApp if it hasn't already taken place.
-     *
-     * @param context Context
-     * @return True - if Firebase is now initialized, False - if the operation could not be completed.
-     */
-    private fun initializeFirebaseIfNeeded(context: Context) {
-        val appName = getAppNameForFirebase(context)
-        val firebaseOptions = FirebaseOptions.Builder()
-            .setGcmSenderId(BootConfig.getBootConfig(context).pushNotificationClientId)
-            .setApplicationId(context.packageName)
-            .build()
-
-        /*
-         * Ensures that Firebase initialization occurs only once for this app. If an exception
-         * isn't thrown, this means that the initialization has already been completed.
-         */
-        try {
-            FirebaseApp.getInstance(appName)
-        } catch (e: IllegalStateException) {
-            SalesforceSDKLogger.i(TAG, "Firebase hasn't been initialized yet", e)
-            FirebaseApp.initializeApp(context, firebaseOptions, appName)
-        }
-    }
-
```

Comment:
None of the private functions in this class were removed, just moved below the public functions.